TLM/TLM/Custom/AI/CustomTouristAI.cs

```diff
@@ -1,181 +1,220 @@
+    using CSUtil.Commons;
+    using JetBrains.Annotations;
+    using Manager.Impl;
+    using RedirectionFramework.Attributes;
+    using State;
+    using State.ConfigData;
+    using Traffic.Enums;
+
+    [TargetType(typeof(TouristAI))]
+    public class CustomTouristAI : TouristAI {
+        [RedirectMethod]
+        [UsedImplicitly]
+        public string CustomGetLocalizedStatus(ushort instanceId, ref CitizenInstance data, out InstanceID target) {
+            string ret = Constants.ManagerFactory.ExtCitizenInstanceManager.GetTouristLocalizedStatus(
+                instanceId,
+                ref data,
+                out bool addCustomStatus,
+                out target);
+
+            // NON-STOCK CODE START
+            if (Options.parkingAI && addCustomStatus) {
+                ret = AdvancedParkingManager.Instance.EnrichLocalizedCitizenStatus(
+                    ret,
+                    ref ExtCitizenInstanceManager.Instance.ExtInstances[instanceId],
+                    ref ExtCitizenManager.Instance.ExtCitizens[data.m_citizen]);
+            }
+
+            // NON-STOCK CODE END
+            return ret;
+        }
+
+        [RedirectMethod]
+        [UsedImplicitly]
+        public VehicleInfo CustomGetVehicleInfo(ushort instanceId,
+                                                ref CitizenInstance citizenData,
+                                                bool forceCar,
+                                                out VehicleInfo trailer) {
 #if DEBUG
-			bool citDebug = (GlobalConfig.Instance.Debug.CitizenInstanceId == 0 || GlobalConfig.Instance.Debug.CitizenInstanceId == instanceID) &&
-				(GlobalConfig.Instance.Debug.CitizenId == 0 || GlobalConfig.Instance.Debug.CitizenId == citizenData.m_citizen) &&
-				(GlobalConfig.Instance.Debug.SourceBuildingId == 0 || GlobalConfig.Instance.Debug.SourceBuildingId == citizenData.m_sourceBuilding) &&
-				(GlobalConfig.Instance.Debug.TargetBuildingId == 0 || GlobalConfig.Instance.Debug.TargetBuildingId == citizenData.m_targetBuilding)
-			;
-			bool debug = GlobalConfig.Instance.Debug.Switches[2] && citDebug;
-			bool fineDebug = GlobalConfig.Instance.Debug.Switches[4] && citDebug;
+            bool citizenDebug = (DebugSettings.CitizenInstanceId == 0
+                                || DebugSettings.CitizenInstanceId == instanceId)
+                               && (DebugSettings.CitizenId == 0
+                                   || DebugSettings.CitizenId == citizenData.m_citizen)
+                               && (DebugSettings.SourceBuildingId == 0
+                                   || DebugSettings.SourceBuildingId == citizenData.m_sourceBuilding)
+                               && (DebugSettings.TargetBuildingId == 0
+                                   || DebugSettings.TargetBuildingId == citizenData.m_targetBuilding);
+            bool logParkingAi = DebugSwitch.BasicParkingAILog.Get() && citizenDebug;
+#else
+            var logParkingAi = false;
```

Comment:
Similar to my earlier comment, do we need this? The callee site where it's used below (line 126) will be culled by the compiler in `RELEASE` builds.